Gems/EMotionFX/Code/EMotionFX/Source/ActorManager.cpp

```diff
@@ -146,60 +149,82 @@ namespace EMotionFX
     Actor* ActorManager::FindActorByName(const char* actorName) const
     {
         // get the number of actors and iterate through them
-        const auto found = AZStd::find_if(m_actors.begin(), m_actors.end(), [actorName](const AZStd::shared_ptr<Actor>& a)
+        const auto found = AZStd::find_if(
+            m_actorAssets.begin(), m_actorAssets.end(),
+            [actorName](const ActorAssetData& a)
         {
-            return a->GetNameString() == actorName;
+                return a.GetAs<Integration::ActorAsset>()->GetActor()->GetNameString() == actorName;
         });
 
-        return (found != m_actors.end()) ? found->get() : nullptr;
+        return (found != m_actorAssets.end()) ? found->GetAs<Integration::ActorAsset>()->GetActor() : nullptr;
```

Comment:
Similarly, I'd expect this to be:
```suggestion
        return (found != m_actorAssets.end()) ? (*found)->GetActor() : nullptr;
```